core/lib/zksync_core/src/state_keeper/io/tests/mod.rs

```diff
@@ -26,119 +24,47 @@ use crate::{
         },
         updates::{MiniblockSealCommand, MiniblockUpdates, UpdatesManager},
     },
-    utils::testonly::prepare_recovery_snapshot,
 };
 
 mod tester;
+mod tests_helpers;
 
 /// Ensure that MempoolIO.filter is correctly initialized right after mempool initialization.
 #[tokio::test]
 async fn test_filter_initialization() {
-    let connection_pool = ConnectionPool::constrained_test_pool(1).await;
-    let tester = Tester::new();
+    let rollup_tester = Tester::new(Arc::new(RollupPubdataPricing {}));
+    let validium_tester = Tester::new(Arc::new(ValidiumPubdataPricing {}));
 
-    // Genesis is needed for proper mempool initialization.
-    tester.genesis(&connection_pool).await;
-    let (mempool, _) = tester.create_test_mempool_io(connection_pool, 1).await;
-
-    // Upon initialization, the filter should be set to the default values.
-    assert_eq!(mempool.filter(), &L2TxFilter::default());
+    tests_helpers::test_filter_initialization(rollup_tester).await;
+    tests_helpers::test_filter_initialization(validium_tester).await;
 }
```

Comment:
We already depend on [test_casing](https://docs.rs/test-casing/latest/test_casing/) crate, so I think you can just reuse it for testing both Rollup/Validium.
You can see the sample usage [here](https://github.com/matter-labs/zksync-era/blob/main/core/lib/zksync_core/src/sync_layer/batch_status_updater/tests.rs#L344).

Reply:
You mean test modules for test-casing purposes? It's fine, I think. The types are cheap and on the first glance it looks better than introducing some kind of shared knowledge between purely test code.